addons/website_sale_comparison/static/src/js/website_sale_comparison.js

```diff
@@ -122,10 +128,12 @@ var ProductComparison = Widget.extend({
             } else {
                 self.load_products([product_id]).then(function(){
                     self.update_content([product_id], false);
+                    def.resolve();
```

Comment:
This deferred is never resolved if we don't go through here, blocking your mutex forever.

> @qsm-odoo done, only saved 6 lines :)

Save 3 more lines and make master-compatible code by removing all of it and just do `return self.load_products(...` and that's it. And that won't be only "9 lines economy", it is master-compatible code, easy to read code and more than 50% code economy on the fix... I am pretty sure it is worth it :)

Reply:
Sure it is ! I was *meh* by introducing that `$.Deferred`. Thanks a lot for your review, much appreciated :smile: